src/auth/auth.validators.ts

```diff
@@ -0,0 +1,32 @@
+import { UnprocessableEntityError } from "@/errors/http";
+import { userService } from "@/user/user.service";
+import bcrypt from "bcryptjs";
+
+export const validateExistingUser = async (email: string) => {
+  const existingUser = await userService.findUserByEmail(email);
+
+  if (existingUser) {
+    throw new UnprocessableEntityError({
```

Comment:
Business logic validators are also a part of the domain layer, thus, one can't use http errors here. Works the same way as in services.